diff --git a/apple/internal/testing/apple_test_assembler.bzl b/apple/internal/testing/apple_test_assembler.bzl
index f613313..63241f6 100644
--- a/apple/internal/testing/apple_test_assembler.bzl
+++ b/apple/internal/testing/apple_test_assembler.bzl
@@ -63,8 +63,6 @@ def _assemble(name, bundle_rule, test_rule, runner = None, runners = None, **kwa
     """
     if runner != None and runners != None:
         fail("Can't specify both runner and runners.")
-    elif not runner and not runners:
-        fail("Must specify one of runner or runners.")

     test_bundle_name = name + ".__internal__.__test_bundle"

