diff --git a/bazel/ceres.bzl b/bazel/ceres.bzl
index d90e5a3..4420565 100644
--- a/bazel/ceres.bzl
+++ b/bazel/ceres.bzl
@@ -202,6 +203,6 @@ def ceres_library(name,
         ],
         visibility = ["//visibility:public"],
         deps = [
-            "@com_github_eigen_eigen//:eigen",
-            "@com_github_google_glog//:glog",
+            "@eigen_archive//:eigen",
+            "@com_github_glog_glog//:glog",
         ],
     )