diff --git a/mediapipe/python/BUILD b/mediapipe/python/BUILD
index 5837be764..8b2ae4541 100644
--- a/mediapipe/python/BUILD
+++ b/mediapipe/python/BUILD
@@ -21,18 +21,13 @@ package(default_visibility = ["//visibility:public"])
 pybind_extension(
     name = "_framework_bindings",
     srcs = ["framework_bindings.cc"],
-    linkopts = select({
-        "//third_party:opencv_source_build": [],
-        "//conditions:default": [
-            "-lopencv_core",
-            "-lopencv_imgproc",
-            "-lopencv_highgui",
-            "-lopencv_video",
-            "-lopencv_features2d",
-            "-lopencv_calib3d",
-            "-lopencv_imgcodecs",
-        ],
-    }),
+    linkopts = [
+        "-L/usr/local/lib64/libopencv_core.a",
+        "-L/usr/local/lib64/libopencv_imgproc.a",
+        "-lm",
+        "-lpthread",
+        "-lstdc++",
+    ],
     module_name = "_framework_bindings",
     deps = [
         ":builtin_calculators",
