diff --git a/third_party/cpuinfo/BUILD.bazel b/third_party/cpuinfo/BUILD.bazel
index 8d89521612..6ea60acdda 100644
--- a/third_party/cpuinfo/BUILD.bazel
+++ b/third_party/cpuinfo/BUILD.bazel
@@ -116,6 +111,8 @@ cc_library(
         ":watchos_x86": COMMON_SRCS + X86_SRCS + MACH_SRCS + MACH_X86_SRCS,
         ":watchos_armv7k": COMMON_SRCS + MACH_SRCS + MACH_ARM_SRCS,
         ":watchos_arm64_32": COMMON_SRCS + MACH_SRCS + MACH_ARM_SRCS,
+        ":tvos_x86_64": COMMON_SRCS + X86_SRCS + MACH_SRCS + MACH_X86_SRCS,
+        ":tvos_arm64": COMMON_SRCS + MACH_SRCS + MACH_ARM_SRCS,
         ":emscripten_wasm": COMMON_SRCS + EMSCRIPTEN_SRCS,
     }),
     copts = C99OPTS + [
@@ -212,7 +209,7 @@ config_setting(
 config_setting(
     name = "ios_armv7",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "ios",
         "cpu": "ios_armv7",
     },
 )
@@ -220,7 +217,7 @@ config_setting(
 config_setting(
     name = "ios_arm64",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "ios",
         "cpu": "ios_arm64",
     },
 )
@@ -228,7 +225,7 @@ config_setting(
 config_setting(
     name = "ios_arm64e",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "ios",
         "cpu": "ios_arm64e",
     },
 )
@@ -236,7 +233,7 @@ config_setting(
 config_setting(
     name = "ios_x86",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "ios",
         "cpu": "ios_i386",
     },
 )
@@ -244,7 +241,7 @@ config_setting(
 config_setting(
     name = "ios_x86_64",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "ios",
         "cpu": "ios_x86_64",
     },
 )
@@ -252,7 +249,7 @@ config_setting(
 config_setting(
     name = "watchos_armv7k",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "watchos",
         "cpu": "watchos_armv7k",
     },
 )
@@ -260,7 +257,7 @@ config_setting(
 config_setting(
     name = "watchos_arm64_32",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "watchos",
         "cpu": "watchos_arm64_32",
     },
 )
@@ -268,7 +265,7 @@ config_setting(
 config_setting(
     name = "watchos_x86",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "watchos",
         "cpu": "watchos_i386",
     },
 )
@@ -276,7 +273,7 @@ config_setting(
 config_setting(
     name = "watchos_x86_64",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "watchos",
         "cpu": "watchos_x86_64",
     },
 )
@@ -284,7 +281,7 @@ config_setting(
 config_setting(
     name = "tvos_arm64",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "tvos",
         "cpu": "tvos_arm64",
     },
 )
@@ -292,7 +289,7 @@ config_setting(
 config_setting(
     name = "tvos_x86_64",
     values = {
-        "crosstool_top": "//tools/osx/crosstool:crosstool",
+        "apple_platform_type": "tvos",
         "cpu": "tvos_x86_64",
     },
 )
