diff --git a/tensorflow_text/core/kernels/BUILD b/tensorflow_text/core/kernels/BUILD
--- a/tensorflow_text/core/kernels/BUILD
+++ b/tensorflow_text/core/kernels/BUILD
@@ -717,7 +717,7 @@
     ],
 )
 
-tf_cc_library(
+cc_library(
     name = "regex_split",
     srcs = ["regex_split.cc"],
     hdrs = ["regex_split.h"],
@@ -1382,7 +1382,7 @@
     ],
 )
 
-tf_cc_library(
+cc_library(
     name = "wordpiece_tokenizer",
     srcs = ["wordpiece_tokenizer.cc"],
     hdrs = ["wordpiece_tokenizer.h"],